internal/controller/nnf_node_storage_controller.go

```diff
@@ -363,12 +364,86 @@ func (r *NnfNodeStorageReconciler) createAllocations(ctx context.Context, nnfNod
 			log.Info("Post activate file system", "allocation", index)
 		}
 
+		// For lustre, PostMount should only happen once. Do this on OST0 only. Additionally, we
+		// need to make sure all of the block storages are ready before we can mount for lustre.
+		if nnfNodeStorage.Spec.FileSystemType == "lustre" && nnfNodeStorage.Spec.LustreStorage.TargetType == "ost" && nnfNodeStorage.Spec.LustreStorage.StartIndex == 0 {
+			ready, err := r.checkForBlockStorageReady(ctx, nnfNodeStorage)
+			if err != nil {
+				return nil, dwsv1alpha2.NewResourceError("could not prep lustre post mount").WithError(err).WithMajor()
+			} else if !ready {
+				// returning here without allocationStatus.Ready=true leads to a requeue
+				return nil, nil
+			}
+
+			// Once the block storages are ready, run lustre postMount
+			ran, err = fileSystem.PostMount(ctx, allocationStatus.Ready)
+			if err != nil {
+				return nil, dwsv1alpha2.NewResourceError("could not run lustre post mount").WithError(err).WithMajor()
+			}
+			if ran {
+				log.Info("Post mount file system (lustre)", "allocation", index)
+			}
+
+			// For other file systems, just run PostMount
+		} else if nnfNodeStorage.Spec.FileSystemType != "lustre" {
+			ran, err = fileSystem.PostMount(ctx, allocationStatus.Ready)
+			if err != nil {
+				return nil, dwsv1alpha2.NewResourceError("could not run post mount").WithError(err).WithMajor()
+			}
+			if ran {
+				log.Info("Post mount file system", "allocation", index)
+			}
+		}
+
 		allocationStatus.Ready = true
 	}
 
 	return nil, nil
 }
 
+// For a given NnfNodeStorage, retrieve all the associated NnfNodeBlockStorages and check if they
+// are all Ready. Obtain the list of NnfNodeBlockStorages by the NnfStorage owner.
+func (r *NnfNodeStorageReconciler) checkForBlockStorageReady(ctx context.Context, nnfNodeStorage *nnfv1alpha3.NnfNodeStorage) (bool, error) {
```

Comment:
I must have mistyped something when describing this.

We won't create any NnfNodeStorages until all of the NnfNodeBlockStorages are ready. You just need to make sure that all of the NnfNodeStorages are ready. There should be one NnfNodeStorage for each Rabbit/target type.

The NnfNodeBlockStorages correspond to all the steps up to making /dev/nvme[*]n[*] devices on the Rabbit.
For Lustre, the NnfNodeStorages correspond to combining those NVMe namespaces with a zpool, and mounting a Lustre target on top (e.g., an OST).

Before making a Lustre client mount we want to make sure that all the Lustre targets are mounted.